sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -103,9 +105,11 @@ class SQLContext(@transient val sparkContext: SparkContext)
    */
   def getAllConfs: immutable.Map[String, String] = conf.getAllConfs
 
+  // TODO how to handle the temp table per user session?
```

Comment:
This is a good question. Ideally we may want to session isolation for temporary tables. However, we can leave this for another PR if you think it makes this PR too complicated. Especially, `HiveMetastoreCatalog` handles both persisted and temporary tables.